pennylane/templates/layers/strongly_entangling.py

```diff
@@ -276,6 +287,55 @@ def imprim_false():
                 pass
 
             rot_loop()
-            cond(n_wires > 1, imprim_true, imprim_false)()
+            cond(num_wires > 1, imprim_true, imprim_false)()
 
         layers()
+
+
+def _strongly_entangling_resources(imprimitive, num_wires, num_layers):
+    resources = {}
+
+    resources[Rot] = num_wires * num_layers
+    if num_wires > 1:
+        resources[imprimitive] = num_wires * num_layers
+
+    return resources
+
+
+@register_resources(_strongly_entangling_resources)
```

Comment:
Thanks for the switch to PL control flow. We should discuss about the use of JAX here though. It is needed to work correctly with program capture, but we should not be using JAX if the new decomposition framework is used without program capture.